code/API_definitions/Application-endpoint-registration.yaml

```diff
@@ -0,0 +1,392 @@
+        content:
+          application/json:
+            schema:
+              type: array
+              items:
+                $ref: "#/components/schemas/ApplicationInstance"
+      responses:
+        "204":
+          $ref: "#/components/responses/NoContent"
+        "401":
+          $ref: "#/components/responses/Unauthorized"
+        "404":
+          $ref: "#/components/responses/NotFound"
+        default:
+          $ref: "#/components/responses/Unexpected"
+      tags:
+        - Application Endpoint Registration
+      summary: Update a application Endpoint
+      description: Update registered application Endpoint information.
+    delete:
+      operationId: deregister-application-endpoint
+      responses:
+        "204":
+          $ref: "#/components/responses/NoContent"
+        "400":
+          $ref: "#/components/responses/BadRequest"
+        "401":
+          $ref: "#/components/responses/Unauthorized"
+        "404":
+          $ref: "#/components/responses/NotFound"
+        default:
+          $ref: "#/components/responses/Unexpected"
+      tags:
+        - Application Endpoint Registration
+      summary: Deregister an application's application Endpoint
+      description: Deregister an application's application
+        Endpoint from the edge cloud zone.
+components:
+  securitySchemes:
+    openId:
+      description: OpenID Provider Configuration Information
+      type: openIdConnect
+      openIdConnectUrl: https://example.com/.well-known/openid-configuration
+  headers:
+    x-correlator:
+      description: Correlation id for the different services
+      schema:
+        type: string
+
+  schemas:
+    ApplicationInstance:
+      description: Application instance represented
+        by application Endpoint definition
+      type: object
+      properties:
+        applicationEndpointsId:
+          $ref: "#/components/schemas/TypesApplicationEndpointsId"
+        applicationEndpoint:
+          $ref: "#/components/schemas/ResourcesapplicationEndpoint"
+        applicationServerProviderName:
+          type: string
+          description: Unique ID representing the Edge
+            Application Provider
+        applicationId:
+          type: string
+          description: Unique ID representing the Edge Application
+        applicationDescription:
+          type: string
+          description: Description of the application Endpoint
+          additionalProperties: false
+        applicationProfileId:
+          $ref: "#/components/schemas/TypesApplicationProfileId"
+        edgeCloudZoneId:
+          $ref: "#/components/schemas/EdgeCloudZoneId"
+        edgeCloudZoneName:
+          $ref: "#/components/schemas/EdgeCloudZoneName"
+        edgeCloudProvider:
+          $ref: "#/components/schemas/EdgeCloudProvider"
+        edgeCloudRegionId:
+          $ref: "#/components/schemas/EdgeCloudRegionName"
+    ResourcesapplicationEndpoint:
+      type: object
```

Comment:
Should we assume that an application can expose only one endpoint or there could be more than one for different purposes? If there can be more than one which only application provider knows then this parameter should allow the multiplicity by having it as an array of objects. Also in that case with each endpoint there should be a discriminator field to allow client applications to know the purpose of a specific endpoint.

Reply:
Although the top level object is an array but then if application has multiple endpoints some of the information will be repeated that many times as they will be common to an application.

Reply:
I also had the same question about repetitive data in the array. For example, is it expected that applicationId is the same for each ApplicationInstance in the array? If any field must be the same for each item in the array, it should be factored out into a single field in a parent object that encapsulates the endpoints array. If every field in the array can be different, then what is the intent of assigning a single applicationEndpointsId to the entire array?